superflore/generators/ebuild/ebuild.py

```diff
@@ -263,34 +256,32 @@ def get_ebuild_text(self, distributor, license_text, die_msg=None):
             ret += "    EPATCH_SOURCE=\"${FILESDIR}\""
             ret += " EPATCH_SUFFIX=\"patch\" \\\n"
             ret += "    EPATCH_FORCE=\"yes\" epatch\n"
-            ret += "ros-cmake_src_prepare\n"
+            ret += "    ros-cmake_src_prepare\n"
             ret += "}\n\n"
 
-        # If we're writing the ebuild for catkin, don't build in binary mode.
-        binary_package = '0' if self.name == 'catkin' else '1'
-
         special_pkgs = ['catkin', 'opencv3', 'stage']
-        catkin_cmake_args = "    local mycmakeargs=(\n"\
-            + "        -DCMAKE_INSTALL_PREFIX=${D%/}${ROS_PREFIX}\n"\
-            + "        -DCMAKE_PREFIX_PATH=${ROS_PREFIX}\n"\
-            + "        -DPYTHON_INSTALL_DIR=lib64/python3.5/site-packages\n"\
+        catkin_cmake_args = "local sitedir=\"$(python_get_sitedir)\"\n"\
+            + "    local mycmakeargs=(\n"\
+            + "        -DCMAKE_INSTALL_PREFIX=/${ROS_PREFIX}\n"\
+            + "        -DCMAKE_PREFIX_PATH=/${ROS_PREFIX}\n"\
+            + "        -DPYTHON_INSTALL_DIR=${sitedir#${EPREFIX}}/${ROS_PREFIX}\n"\
```

Comment:
Sure that is not `PYTHON_INSTALL_DIR=${ROS_PREFIX}${sitedir#${EPREFIX}}`?

I'm experiencing strange things with this, not all packages respect this variable (roslz4)
For example with roslz4:

it install pythons (`.py` and company) in `/opt/ros/usr/lib64/python2.7/site-packages/`
but it install libraries (`.so`) in `/usr/lib64/python2.7/site-packages/`

if I prefix `/opt/ros/` it became absurdly

pythons (`.py` and company) in `/opt/ros/opt/ros/usr/lib64/python2.7/site-packages/`
libraries (`.so`) in `/opt/ros/usr/lib64/python2.7/site-packages/`

@aballier what is the correct way to install pythons in an offset directory?
@aballier I'm trying to get the `ros-catkin.eclass` working with gentoo's ros but I'm stuck with this issue, where we can discuss further?